create_jenkins_job.py

```diff
@@ -194,6 +194,16 @@ def create_job(os_name, job_name, template_file, additional_dict):
                 'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
             })
 
+        # configure nightly job for testing rmw/rcl based packages with address sanitizer on linux
+        if os_name == 'linux':
+            create_job(os_name, 'nightly_' + os_name + '_address_sanitizer', 'ci_job.xml.em', {
+                'cmake_build_type': 'Debug',
+                'time_trigger_spec': PERIODIC_JOB_SPEC,
+                'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
+                'build_args_default': data['build_args_default'] + ' --mixin asan --packages-up-to test_communication test_rclcpp',
+                'test_args_default': '--event-handlers console_direct+ --executor sequential --packages-select test_communication test_rclcpp',
```

Comment:
Maybe I missed it in the discussion but why should the testing be limited to these two packages?

Reply:
My understanding was that @bhatsach was targeting a limited set of packages to avoid `osrf_memory_testing` related issues with the goal of expending support down the road once a technical solution is found for those packages.

Reply:
@nuclearsandwich helped prune the packages to be tested as seen in the commit: https://github.com/ros2/ci/pull/245/commits/0f502cc7a72ff6284aab59c7f7d7ca359614c3ea. Our intention was to avoid unnecessary packages so that build and test both are reduced to a reasonable size and time.

Reply:
I am not sure what labels all other packages as unnecessary but I get the point of reducing the runtime of the job - it just seems like an arbitrary trade off. Anyway I don't mind.

Reply:
> I am not sure what labels all other packages as unnecessary

Unnecessary is perhaps too strong a word, but yes the intent is to limit runtime and focus on the communications packages. Where @bhatsach's approach was to grab every package likely to be part of the communications layer, I started with few test packages and plan to grow out.

I'm thinking about moving the package selection to another mixin so its easier to edit it and reproduce the build locally.